packit/upstream.py

```diff
@@ -856,3 +858,35 @@ def _get_rpms_from_rpmbuild_output(self, output: str) -> List[str]:
             )
 
         return rpms
+
+    @staticmethod
+    def _template2regex(template) -> str:
+        """
+        Converts tag template to regex with named groups.
+
+        :param template: tag_template string
+        :return: regex string which can be used by python re module
+        """
+
+        p = re.compile(r"{(.*?)}")
+        return p.sub(r"(?P<\g<1>>.*)", template)
+
+    def get_version_from_tag(self, tag: str) -> str:
+        """
+        Extracts version from git tag using upstream_template_tag
+
+        :param tag: git tag containing version
+        :return: version string
+        """
+
+        field = "version"
+        regex = self._template2regex(self.package_config.upstream_tag_template)
+        p = re.compile(regex)
+        try:
+            return p.match(tag).group(field)
+        except (IndexError, AttributeError):
+            logger.error(
+                f'Unable to extract "{field}" from {tag} using '
+                f"{self.package_config.upstream_tag_template}"
+            )
+            raise
```

Comment:
I am looking for the time we can be python>=3.8 only and use [Assignment Expressions](https://www.python.org/dev/peps/pep-0572/).

But I would still try to avoid unnecessary `try-except` when not needed...

```suggestion
        match = p.match(tag)
        if match and field in match.groups():
            return match.group(field)
        else:
            msg = (
                f'Unable to extract "{field}" from {tag} using '
                f"{self.package_config.upstream_tag_template}"
            )
            logger.error(msg)
            raise PackitException(msg)
```

Reply:
fixed

why you prefer if...else over try...except when dealing with unwanted errors?